lib/puppet/util/run_mode.rb

```diff
@@ -55,11 +55,19 @@ def which_dir( system, user )
 
     class UnixRunMode < RunMode
       def conf_dir
-        which_dir("/etc/puppet", "~/.puppet")
+        which_dir("/etc/puppetlabs/agent", "~/.puppet")
       end
 
       def var_dir
-        which_dir("/var/lib/puppet", "~/.puppet/var")
+        which_dir("/opt/puppetlabs/agent/cache", "~/.puppet/var")
+      end
+
+      def run_dir
+        which_dir("/var/run/puppetlabs", "~/.puppet/var/run")
+      end
+
+      def log_dir
+        which_dir("/var/log/puppetlabs/agent", "~/.puppet/var/log/")
```

Comment:
Talked with @nwolfe and we discovered that the problem now is that if you specify `--vardir` as an argument, you now also need to specifiy `rundir` and `logdir`, as they no longer fall under `vardir` dynamically.